Sources/Networking/AlamofireNetworkClient/OAuth/OAuthInterceptor.swift

```diff
@@ -13,27 +13,28 @@ import PovioKit
 public struct OAuthContainer {
   public let accessToken: String
   public let refreshToken: String
-  public let expirationDate: Date
   
   public init(
     accessToken: String,
-    refreshToken: String,
-    expirationDate: Date
+    refreshToken: String
   ) {
     self.accessToken = accessToken
     self.refreshToken = refreshToken
-    self.expirationDate = expirationDate
   }
 }
 
 public protocol OAuthStorage: AnyObject {
   var accessToken: String? { get set }
   var refreshToken: String? { get set }
-  var accessTokenExpirationDate: Date? { get set }
 }
 
 public protocol OAuthProvider {
   func refresh(with refreshToken: String) -> Promise<OAuthContainer>
+  func isAccessTokenValid() -> Bool
+}
+
+public extension OAuthProvider {
+  func isAccessTokenValid() -> Bool { true }
```

Comment:
So if we don't override it, token is always "valid"? 🤔

Reply:
If we provide a default implementation then `false` doesn't make any sense. So we either remove the default implementation or keep it as is.
The reason for default implementation is that most of the time the app is not responsible to check whether the token is valid or not. That is the responsibility of the server.
If you actually have to perform some logic then it is up to you to implement it.

Reply:
Actually, let me think a bit more about this. It might be best to just remove the default implementation and not rely on any assumptions.